server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/SieveScriptRoutes.java

```diff
@@ -0,0 +1,187 @@
+    public SieveScriptRoutes(SieveRepository sieveRepository, UsersRepository usersRepository) {
+        this.sieveRepository = sieveRepository;
+        this.usersRepository = usersRepository;
+    }
+
+    @Override
+    public String getBasePath() {
+        return ROOT_PATH;
+    }
+
+    @Override
+    public void define(Service service) {
+        defineAddActiveSieveScript(service);
+    }
+
+    @PUT
+    @ApiOperation(value = "Upload a new Sieve Script")
+    @Path(value = ROOT_PATH + "/{" + USER_NAME + "}/" + SCRIPTS + "/{" + SCRIPT_NAME + "}")
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "OK"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid username"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid Sieve script name"),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Empty script is not accepted"),
+        @ApiResponse(code = HttpStatus.NOT_FOUND_404, message = "User not exist")
+    })
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            name = USER_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Username"),
+        @ApiImplicitParam(
+            name = SCRIPT_NAME,
+            required = true,
+            paramType = "path",
+            dataType = "String",
+            example = ROOT_PATH + "/userNameA/" + SCRIPTS + "/scriptName1",
+            value = "Script name"),
+        @ApiImplicitParam(
+            required = false,
+            paramType = "query parameter",
+            dataType = "Boolean",
+            defaultValue = "False",
+            example = "?activate=true",
+            value = "If present, automatically activating the script.")
+    })
+    public void defineAddActiveSieveScript(Service service) {
+        service.put(USER_SCRIPT_PATH, this::addActiveSieveScript);
+    }
+
+    private HaltException addActiveSieveScript(Request request, Response response) throws UsersRepositoryException, QuotaExceededException, StorageException, ScriptNotFoundException {
+        User user = extractUser(request);
+        ScriptName script = extractScriptName(request);
+        sieveRepository.putScript(user, script, extractSieveScriptFromRequest(request));
+        if (isActivated(request.queryParams(ACTIVATE_PARAMS))) {
+            sieveRepository.setActive(user, script);
+        }
+        return halt(HttpStatus.NO_CONTENT_204);
+    }
+
+    private User extractUser(Request request) throws UsersRepositoryException {
+        Optional<String> userName = Optional.ofNullable(request.params(USER_NAME));
+        if (!userName.isPresent() || Strings.isNullOrEmpty(userName.get().trim())) {
+            throw400withInvalidArgument("Invalid username");
+        }
+
+        if (!usersRepository.contains(userName.get())) {
```

Comment:
please, never call `.get` on an optional several time, once it's validated as not empty, please keep the content of it for the remaining of the code (my orElse comment would allow doing that safely)